compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToUKernels.cpp

```diff
@@ -0,0 +1,296 @@
+  AffineMap map1 = cast<AffineMapAttr>(indexingMaps[1]).getValue();
+  AffineMap map2 = cast<AffineMapAttr>(indexingMaps[2]).getValue();
+
+  if (map0.getNumResults() != 6 || map1.getNumResults() != 6 ||
+      map2.getNumResults() != 6 || map0.getNumInputs() != 9 ||
+      map1.getNumInputs() != 9 || map2.getNumInputs() != 9) {
+    return false;
+  }
+
+  // Since this is invokved after the final pack-and-transpose we need to
+  // extract dimensions for M*K*k*m*m0*k0 x K*N*n*k*k0*n0 -> M*N*n*m*m0*n0.
+  AffineExpr M = map2.getResult(0);
+  AffineExpr N = map2.getResult(1);
+  AffineExpr n = map2.getResult(2);
+  AffineExpr m = map2.getResult(3);
+  AffineExpr m0 = map2.getResult(4);
+  AffineExpr n0 = map2.getResult(5);
+  AffineExpr K = map0.getResult(1);
+  AffineExpr k = map0.getResult(2);
+  AffineExpr k0 = map0.getResult(5);
+
+  auto *context = indexingMaps.getContext();
+  auto mapA =
+      AffineMapAttr::get(AffineMap::get(9, 0, {M, K, k, m, m0, k0}, context));
+  auto mapB =
+      AffineMapAttr::get(AffineMap::get(9, 0, {K, N, n, k, k0, n0}, context));
+  auto mapC =
+      AffineMapAttr::get(AffineMap::get(9, 0, {M, N, n, m, m0, n0}, context));
+  auto maps = ArrayAttr::get(context, {mapA, mapB, mapC});
+  return indexingMaps == maps;
+}
+
+/// Matches a linalg.generic operation which is basically a tiled matmul and
+/// converts it into a iree_codegen.ukernel."iree_amdaie_uk_matmul" operation,
+/// that is later lowered into a call to the microkernel.
+static FailureOr<IREE::Codegen::UKernelOpInterface> matchDAGForUKernel(
+    RewriterBase &rewriter, linalg::GenericOp op, std::string ukernelName) {
+  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(op);
+  if (!hasUkernel(targetAttr, ukernelName)) {
+    return failure();
+  }
+  Value lhs = op.getDpsInputOperand(0)->get();
+  Value rhs = op.getDpsInputOperand(1)->get();
+  Value out = op.getDpsInitOperand(0)->get();
+  auto outType = llvm::cast<ShapedType>(out.getType());
+  Type lhsElemType = llvm::cast<ShapedType>(lhs.getType()).getElementType();
+  Type rhsElemType = llvm::cast<ShapedType>(rhs.getType()).getElementType();
+  Type outElemType = outType.getElementType();
+  if (!(lhsElemType.isBF16() && rhsElemType.isBF16()) ||
+      !(outElemType.isF32() || outElemType.isBF16())) {
+    return rewriter.notifyMatchFailure(
+        op, "unsupported combination of element types for microkernel");
+  }
+
+  // Check if the accumulator is zero-filled.
+  if (isInitializedToZero(out)) {
+    // Here the matmul ukernel op won't read the existing accumulator, so its
+    // defining op can be discarded.
+    if (auto fillOp = out.getDefiningOp<linalg::FillOp>()) {
+      out = fillOp.getDpsInitOperand(0)->get();
+    }
+  }
+
+  Location loc = op.getLoc();
+  // Since this is invokved after the final pack-and-transpose we need to
+  // extract dimensions for M*K*k*m*m0*k0 x K*N*n*k*k0*n0 -> M*N*n*m*m0*n0.
+  Value M = rewriter.create<tensor::DimOp>(loc, lhs, 0);
```

Comment:
We dont need to pass all this information. We need to pass only the tensor dimensions that are needed by the ukernel implementation. Shouldnt matter much, but just seems unnecessary.